chainlibpy/transaction.py

```diff
@@ -2,31 +2,127 @@
+
+            client (GrpcClient): GrpcClient object to connect to chain
+
+            gas_limit (int, optional): maximum gas can be used in transaction processing.
+            Defaults to DEFAULT_GAS_LIMIT.
+
+            fee (Optional[List[Coin]], optional): amount of coins to be paid as a fee.
+            Defaults to None.
+
+            memo (str, optional): note to be added to the transaction. Defaults to "".
+
+            timeout_height (int, optional): this transaction will not be processed
+            after timeout height. Defaults to None.
+        """
+        self._chain_id = chain_id
+        self._packed_msgs = self._pack_msgs_to_any_msgs(msgs)
+        self._fee = fee
+        self._from_wallets = from_wallets
+        self._memo = memo
+        self._timeout_height = timeout_height
+        self._account_number = account_number
+        self._gas_limit = gas_limit
+        self._client = client
+
+    def _pack_msgs_to_any_msgs(self, msgs: List[message.Message]) -> List[any_pb2.Any]:
+        return [pack_to_any_message(msg) for msg in msgs]
+
+    def append_message(self, *msgs: message.Message) -> "Transaction":
+        """Append more messages in this transaction.
+
+        Args:
+            *msgs (message.Message): messages to be included in this transaction
+
+        Returns:
+            Transaction: transaction object with newly added messages
+        """
+        self._packed_msgs.extend(self._pack_msgs_to_any_msgs(list(msgs)))
+
+        return self
+
+    def get_tx_body(self) -> TxBody:
+        """Get cosmos.tx.v1beta1.TxBody protobuf message of this transaction.
+
+        Returns:
+            TxBody: cosmos.tx.v1beta1.TxBody message
+        """
+        return TxBody(messages=self._packed_msgs, memo=self._memo)
+
+    def get_auth_info(self) -> AuthInfo:
+        signer_infos = []
+        for wallet in self._from_wallets:
+            # query account to get the latest account.sequence
+            account = self._client.query_account(wallet.address)
+
+            signer_info = SignerInfo(
+                public_key=pack_to_any_message(PubKey(key=wallet.public_key)),
+                mode_info=ModeInfo(single=ModeInfo.Single(mode=SignMode.SIGN_MODE_DIRECT)),
+                sequence=account.sequence,
+            )
+
+            signer_infos.append(signer_info)
+
+        return AuthInfo(
+            signer_infos=signer_infos, fee=Fee(amount=self._fee, gas_limit=self._gas_limit)
+        )
+
+    def get_sign_doc(self) -> SignDoc:
+        return SignDoc(
+            body_bytes=self.get_tx_body().SerializeToString(),
+            auth_info_bytes=self.get_auth_info().SerializeToString(),
+            chain_id=self._chain_id,
+            account_number=self._account_number,
+        )
+
+    def get_signed_tx(self, signatures: List[bytes]) -> Tx:
+        return Tx(body=self.get_tx_body(), auth_info=self.get_auth_info(), signatures=signatures)
```

Comment:
maybe we should make it as same as the js lib? redefine the `Tx`, add some function like `set_signature` `to_signed `?